tests/frontend/test_GameGUI.py

```diff
@@ -1,64 +1,58 @@
-import unittest
+import pytest
 from unittest.mock import patch
 import pygame as pg
 from frontend.GameGUI import GameGUI
 
-'''
-Setup: The setUp method initializes the GameGUI object before each test. It uses patch decorators to mock pygame modules, since rendering and handling real-time events aren't suitable for unit tests.
 
-test_check_events_exit: This test verifies that the check_events() method handles the QUIT event correctly by calling pygame.quit() and sys.exit().
+@pytest.fixture(scope='function')
+def game_fixture():
+    """Fixture to initialize and clean up the GameGUI instance for each test."""
+    # Initialize Pygame display to prevent errors with display functions
+    pg.display.set_mode([1, 1])  # Minimal size to initialize display
+    game = GameGUI()  # Create instance of GameGUI
+    yield game  # Provide the game instance to the test
+    pg.quit()  # Quit Pygame after each test
 
-test_check_events_no_quit: It ensures that no exit actions are taken when QUIT is not in the events.
 
-test_run_game_quit: Tests that the run_game() method calls build_board() once and exits gracefully upon a QUIT event.
+def test_check_events_exit(game_fixture):
+    game = game_fixture
 
-test_run_game: Verifies that get_cell_clicked() is called multiple times as expected during the game loop. It simulates a series of events (KEYDOWN and QUIT) to control the game loop flow.
-
-'''
-class TestGameGUI(unittest.TestCase):
```

Comment:
How come you chose to not have any of the tests be in a class? ALl of our other tests are in classes.

Reply:
Seconding this, but because we have to list classes on the documentation -- the tests need to be in classes 😅